cirq/google/op_deserializer.py

```diff
@@ -158,3 +196,86 @@ def _args_from_proto(
             if value is not None:
                 return_args[arg.constructor_arg_name] = value
         return return_args
+
+
+class CircuitOpDeserializer(OpDeserializer):
+    """Describes how to serialize CircuitOperations."""
+
+    @property
+    def serialized_id(self):
+        return 'circuit'
+
+    def from_proto(
+        self,
+        proto: v2.program_pb2.CircuitOperation,
+        *,
+        arg_function_language: str = '',
+        constants: List[v2.program_pb2.Constant] = None,
+        raw_constants: List[Any] = None,
+    ) -> 'cirq.CircuitOperation':
+        """Turns a cirq.google.api.v2.CircuitOperation proto into a CircuitOperation."""
+        if constants is None or raw_constants is None:
```

Comment:
Should we also check if len(raw_constants) > proto.circuit_constant_index?

Reply:
Done.